CMakeLists.txt

```diff
@@ -87,7 +117,9 @@ if (MSVC)
 endif ()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wunused-parameter -Wextra -Wreorder")
+    if(NOT EMSCRIPTEN)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wunused-parameter -Wextra -Wreorder")
+    endif()
     
     CHECK_CXX_COMPILER_FLAG("-std=c++17" HAS_CPP_17_FLAG)
     if (HAS_CPP_17_FLAG)
```

Comment:
@DerThorsten not sure what I'm overlooking/missing here :|
Could you maybe take a quick look ?!

Reply:
I had a similar issue recently but I forgot what the solution was.
What happens if you drop the check and just add the c++17 flag for each target?

Reply:
Trying that out in the next commit !

Reply:
Hmm, I think everything builds as expected. Looks like there is another error though :(
```
cache:INFO: generating system library: sysroot/lib/wasm32-emscripten/pic/libsockets.a... (this will be cached in "/home/runner/micromamba/envs/xeus-r-wasm-build/lib/python3.12/site-packages/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/pic/libsockets.a" for subsequent builds)
system_libs:INFO: compiled 21 inputs
cache:INFO:  - ok
LLD 18.0.0
/home/runner/micromamba/envs/xeus-r-wasm-build/lib/python3.12/site-packages/emsdk/upstream/bin/llvm-objcopy: error: 'xr.wasm': No such file or directory
em++: error: '/home/runner/micromamba/envs/xeus-r-wasm-build/lib/python3.12/site-packages/emsdk/upstream/bin/llvm-objcopy xr.wasm xr.wasm --remove-section=.debug* --remove-section=producers' failed (returned 1)
make[2]: *** [CMakeFiles/xr.dir/build.make:106: xr.js] Error 1
```

Reply:
I think all this logic can be removed now in favor of using `target_compile_features(${target_name} PRIVATE cxx_std_17)` in the macro `macro(xeus_r_set_common_options target_name)`.